### ISSUE TITLE
demo() and code showing bug

### ISSUE THREAD

**Issue by pasqualedem** [massgh/ipyslides#2]
When running:
```python
import ipyslides as isd
ls = isd.LiveSlides().demo()
```
I get the following error:

```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Input In [1], in <module>
      1 import ipyslides as isd
----> 2 ls = isd.LiveSlides().demo()
      3 ls

File ~\.conda\envs\pymc_env\lib\site-packages\ipyslides\_base\base.py:157, in BaseLiveSlides.demo(self)
    155 self._check_computed('load demo')
    156 get_ipython().user_ns['_s_l_i_d_e_s_'] = self
--> 157 from .. import _demo
    158 slides = _demo.slides # or it is self
    159 with slides.slide(100):

File ~\.conda\envs\pymc_env\lib\site-packages\ipyslides\_demo.py:81, in <module>
     78                 write([slides.doc(write,'LiveSlides'), slides.doc(iwrite,'LiveSlides')])
     79                 write("#### If an object does not render as you want, use `display(object)` or it's own library's mehod to display inside Notebook.")
---> 81             write(s.show_lines([0,1]))
     82 # Matplotlib
     83 with slides.slide(8,background='linear-gradient(to right, #FFDAB9 0%, #F0E68C 100%)'):

File ~\.conda\envs\pymc_env\lib\site-packages\ipyslides\source.py:41, in _Source_Widget.show_lines(self, lines)
     38     raise TypeError(f'lines must be list, tuple or range, not {type(lines)}')
     40 start, *middle = self._code.split('<code>')
---> 41 middle[-1], end = middle[-1].split('</code>')
     42 middle[-1] += '</code>'
     43 _max_index = len(middle) - 1

IndexError: list index out of range
```
Some slides of the demo are shown, but not all. Also in some slides the code cells are empty, the same happens when I try to show code with source.context():

```python
%%slide 4
with ls.source.context() as s:
    import numpy as np, matplotlib.pyplot as plt
    x = np.linspace(0,2*np.pi)
    with plt.style.context('ggplot'):
        fig, ax = plt.subplots(figsize=(3.4,2.6))+
        _ = ax.plot(x,np.cos(x))
ls.write('## Plotting and displaying code')
ls.write(ax,s)
```

**Comment by asaboor-gh** [comment]
There was a bug left in a version. You can either update to 1.4.3 or do the following in current version 
```python
import ipyslides as isd
ls = isd.LiveSlides() # create first
ls.show() # showing it enable many things when displayed.
ls.demo() # now it should work.
``` 
Test this modification on [binder](https://mybinder.org/v2/gh/massgh/ipyslides-voila/HEAD?urlpath=lab%2Ftree%2Fnotebooks%2Fipyslides.ipynb)
In any case, updating to latest version will fix bugs

**Comment by asaboor-gh** [comment]
```python
%%slide 4
with ls.source.context() as s:
    import numpy as np, matplotlib.pyplot as plt
    x = np.linspace(0,2*np.pi)
    with plt.style.context('ggplot'):
        fig, ax = plt.subplots(figsize=(3.4,2.6))+
        _ = ax.plot(x,np.cos(x))
ls.write('## Plotting and displaying code')
ls.write(ax,s)
```
I see a plus sign at end of line fig, ax .. that may be causing bug

**Comment by pasqualedem** [comment]
> 

The plus sign is a typo added later, however I tried different times, with different codes, and it shows only an empty code cells.

**Comment by asaboor-gh** [comment]
> > 
> 
> The plus sign is a typo added later, however I tried different times, with different codes, and it shows only an empty code cells.

Is this issue with only this context or others have issue too. I will exactly run this snippet of code and let you know

**Comment by asaboor-gh** [comment]
![Screenshot_20220322-161953_Chrome](https://user-images.githubusercontent.com/48522862/159568846-68a590e8-6d43-409f-ad6f-8c6a79ce597e.png)
It is working on my side. You may need to update

**Comment by pasqualedem** [comment]
I have updated to 1.4.4 and still have it, maybe is the Jupyter-lab version, I'm on 3.3.0

**Comment by asaboor-gh** [comment]
I checked Jupyterlab 3.0 - 3.3 and they are working well. Binder link is working on 3.3 version currently and you can check it yourself. 

Can you tell me the version of Python and IPython you are using? That is more of a concern for this `source.context` method. 
Also try to delete cache when updating a package with `pip install ipyslides --no-cache-dir` option

**Comment by pasqualedem** [comment]
I'm on Python 3.10

**Comment by asaboor-gh** [comment]
> I'm on Python 3.10

Here I checked upto Python 3.9 and it works. Looks like python 3.10 is not yet available on Binder.
![image](https://user-images.githubusercontent.com/48522862/159778103-3243c9c6-062a-4bc0-880f-957bf8808ede.png)

I will check 3.10 as well on my own. But there are other options to show code as well. like:
```python
%%slide 4
import numpy as np, matplotlib.pyplot as plt
x = np.linspace(0,2*np.pi)
with plt.style.context('ggplot'):
    fig, ax = plt.subplots(figsize=(3.4,2.6))+
    _ = ax.plot(x,np.cos(x))

ls.write('## Plotting and displaying code')

source = ls.source.from_string('''
import numpy as np, matplotlib.pyplot as plt
x = np.linspace(0,2*np.pi)
with plt.style.context('ggplot'):
    fig, ax = plt.subplots(figsize=(3.4,2.6))+
    _ = ax.plot(x,np.cos(x))
''')

ls.write(ax, source)
```
I am planning to add a code execution option in markdown blocks (like R in rmarkdown) in the next version as well. Till then, this way works.

**Comment by asaboor-gh** [comment]
> I'm on Python 3.10

That bug is fixed in 1.4.5 with additional changes. Now it works.
